### PR TITLE
Get Production Tracking Aspect Model

### LINKED ISSUE #546: [New Model]: GetProductionTrackingData
## Model Description

The Shop Floor Information Service will be extended in 2405 by a method to exchange process data collected during the production of a process for tracebility or documentation. This service will have two actions: a request asking for the data and the answer with the requested data. 

## Relevant Standards

None

This Request corresponds to the GetProductionTrackingData Request:

## Example Data

| Level |  Property | mand./opt. | Semantic ID |
|----|----|----|----|
|CX-Header|"senderBpn"|mandatory|string|
|CX-Header|recipientBpn|mandatory|string|
|CX-Header|context|mandatory|string|
|CX-Header|messageId|mandatory|UUID|
|CX-Header|sentDateTime|mandatory|string|
|CX-Header|version|mandatory|string|
|SIS-Payload|versionDataModel|mandatory|string|
|SIS-Payload|customerId|mandatory|string|
|SIS-Payload|orderId|mandatory|string|
|SIS-Payload|positionId|mandatory|string|
|SIS-Payload|identifierType||[serial, batch, bom]|mandatory |[serial]|[serial]|
|SIS-Payload|identifierNumber|mandatory|string|
|SIS-Payload|processReferenceType||[processStep, capability]|
|SIS-Payload|bopId|0..1|BoP||BoP_ID|
|SIS-Payload|processStepIdentifierList|1|
|SIS-Payload|  capabilityId|(0..n)|
||      partInstanceLevel||[bom, serial]|
||      bomId|1|      BoM|
||      bomElementId|1|      BoM-Element|
||      serialNumber|1|      Seriennummer|
|SIS-Payload|  processStepId|(0..n)|"ProcessStepIdentifier|
|SIS-Payload|     processParameterList|1||
|SIS-Payload|       processParameterName|1..n|"ProcessParameter[string]|
||       processParameterSemanticID|1..n|SemanticID|

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by FlorianDue**
## Description

New aspect model for the production tracking request within the SopfloorInformationService
 -->

Closes #546 

## MS2 Criteria
(to be filled out by PR reviewer)
- [ ] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [ ] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [ ] the identifiers for all model elements **start with a capital letter** except for properties
- [ ] the identifier for **properties starts with a small letter**
- [ ] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [ ] Property and the referenced Characteristic should not have the same name
- [ ] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [ ] use **abbreviations only when necessary** and if these are sufficiently common
- [ ] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [ ] fields `preferredName` and `description` are not the same
- [ ] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [ ] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [ ] units are referenced from the SAMM unit catalog whenever possible
- [ ] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [ ] when relying on **external standards**, they are referenced through a **"see"** element
- [ ] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [ ] metadata.json exists with status "release"
- [ ] generated json schema validates against example json payload
- [ ] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [ ] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [ ] All required reviewers have approved this PR (see reviewers section)
- [ ] The new aspect (version) will be implemented by at least one data provider
- [ ] The new aspect (version) will be consumed by at least one data consumer
- [ ] There exists valid test data
- [ ] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [ ] The model has at least version '1.0.0'
- [ ] If a previous model exists, model deprecation has been checked for previous model
- [ ] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+##########################################################################################
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.0.0#>.
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.0.0#>.
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.0.0#>.
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.0.0#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix : <urn:samm:io.catenax.shopfloor_information.productionTrackingRequest:1.0.0#>.
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:24]: please correct namespace

Comment (FlorianDue) [reply io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl]: Thank you fo your feedback!

Namespace name is corrected

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    samm:dataType xsd:string;
+    samm-c:values ("bom" "serial").
+
+:ProcessPartameterListCharacteristic a samm-c:List;
+    samm:preferredName "ProcessPartameterListCharacteristic"@en;
+    samm:description "List os process parameters, whose values the requester wants to receive from the provider "@en;
+    samm:dataType :ProcessParameterRequestType.
+
+:ProcessParameterRequestType a samm:Entity;
+    samm:preferredName "ProcessParameterRequestType"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:130]: `description` missing

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    samm:dataType xsd:string.
+
+:ProcessReferenceTypeCharacteristic a samm-c:Enumeration;
+    samm:preferredName "ProcessReferenceTypeCharacteristic"@en;
+    samm:description "Enumeration to distinguish between a request based on process steps or a request based on capabilities"@en;
+    samm:dataType xsd:string;
+    samm-c:values ("processStep" "capability").
+
+:ProcessStepIdentifierCharacteristic a samm-c:List;
+    samm:preferredName "ProcessStepIdentifierCharacteristic"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:70]: `description` missing

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+:GetProductionTrackingData a samm:Aspect;
+    samm:preferredName "GetProductionTrackingData"@en;
+    samm:properties (:customerId [ samm:property :billOfProcessId; samm:optional "true"^^xsd:boolean ] :processReferenceType :processStepIdentifierList ext-header2:header [ samm:property ext-part:catenaXId; samm:optional "true"^^xsd:boolean ] ext-part:localIdentifiers ext-header2:version);
+    samm:operations ();
+    samm:events ();
+    samm:description "Aspect to request product specific data from modular production"@en.
+
+:customerId a samm:Property;
+    samm:preferredName "customerId"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:37]: `customerId` should not be in camel case. it should be in plain English.

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    samm:description "internal customer Id"@en;
+    samm:characteristic ext-number:BpnlTrait.
+
+:billOfProcessId a samm:Property;
+    samm:preferredName "billOfProcessId"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:42]: please correct `preferredName`

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#>.
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.0.0#>.
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.0.0#>.
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.0.0#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix : <urn:samm:io.catenax.shopfloor_information.get_production_tracking:1.0.0#>.
+@prefix ext-header2: <urn:samm:io.catenax.shared.message_header:2.0.0#>.
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#>.
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTrackingData.ttl:26]: please consider using business_partner_number 2.0.0 shared model here instead of 1.0.0

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#>.
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#>.
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#>.
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix : <urn:samm:io.catenax.shopfloor_information.get_production_tracking:1.0.0#>.
+@prefix ext-header2: <urn:samm:io.catenax.shared.message_header:2.0.0#>.
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#>.
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl:25]: business_partner_number 2.0.0 should be used.

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    samm:exampleValue "processStep".
+
+:billOfProcessId a samm:Property;
+    samm:preferredName "Bill of Process Identifier"@en;
+    samm:description "Id to identify an instance of the bill of process data model "@en;
+    samm:characteristic :StringCharacteristic;
+    samm:exampleValue "12345TestBoP.org".
+
+:customerId a samm:Property;
+    samm:preferredName "customerId"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl:68]: remove camel case from `customerId`

File: io.catenax.shopfloor_information.get_production_tracking/RELEASE_NOTES.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+All notable changes to this model will be documented in this file.
+
+## [1.0.0] 29.01.2024
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/RELEASE_NOTES.md:5]: you can change the date to 5th Feb if you want MS3 today.

Comment (FlorianDue) [reply io.catenax.shopfloor_information.get_production_tracking/RELEASE_NOTES.md]: corrections are made. Just as an addition information. Both modesl, this and #556  use the serial_part model version 3.0.0, which is not merged yet.

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl [hunk 10]
```diff
@@ -0,0 +1,178 @@
+##########################################################################################
+# Copyright (c) 2023-2024 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V. (represented by Fraunhofer IOSB & Fraunhofer IWU & Fraunhofer IPA)
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl:2]: please also change the year to 2024 only.

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl (hunk 10 above)
Comment (FlorianDue) [reply io.catenax.shopfloor_information.get_production_tracking/1.0.0/GetProductionTracking.ttl]: update is pushed, thank you for the review!
